install-paperless-ngx.sh

```diff
@@ -3,16 +3,16 @@
 ask() {
 	while true ; do
 		if [[ -z $3 ]] ; then
-			read -p "$1 [$2]: " result
+			read -r -p "$1 [$2]: " result
 		else
-			read -p "$1 ($3) [$2]: " result
+			read -r -p "$1 ($3) [$2]: " result
 		fi
 		if [[ -z $result ]]; then
 			ask_result=$2
 			return
 		fi
 		array=$3
-		if [[ -z $3 || " ${array[@]} " =~ " ${result} " ]]; then
+		if [[ -z $3 || " ${array[*]} " =~ ${result} ]]; then
```

Comment:
Does this really work? I don't understand why there were spaces before and after but it looks like they are gone now.

Reply:
From the docs I read, it should but that needs to be tested

Regards

Schnuffle

Reply:
Here's the log form pre-commit:
In install-paperless-ngx.sh line 15:
if [[ -z $3 || " ${array[@]} " =~ " ${result} " ]]; then
_________ ^-------------^ SC2199 (error): Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).
________________________^-----------^ SC2076 (warning): Remove quotes from right-hand side of =~ to match as a regex rather than literally.

Reply:
@schnuffle could you test that this still works? Same with the other instance below.

Reply:
I tested this out and my choice was correctly validated.

```
Database backend (postgres sqlite) [postgres]: test
Invalid option: test
Database backend (postgres sqlite) [postgres]: sqlite3
Invalid option: sqlite3
Database backend (postgres sqlite) [postgres]: mariadb
Invalid option: mariadb
Database backend (postgres sqlite) [postgres]: sqlite
```